win32/src/winapi/user32/message.rs

```diff
@@ -268,13 +279,13 @@ pub async fn GetMessageA(
         }
     }
 
-    let msg = lpMsg.unwrap();
-    *msg = machine.state.user32.messages.pop_front().unwrap();
-    if !hWnd.is_null() && (!msg.hwnd.is_null() && msg.hwnd != hWnd) {
-        todo!("message for other window");
-    }
-    if msg.message == WM::QUIT as u32 {
-        return 0;
+    if let Some(index) = find_message(machine, hWnd, wMsgFilterMin, wMsgFilterMax) {
```

Comment:
One thing I had worried about here is if the program calls GetMessage with filtermin/max and there isn't an available message that matches, I think GetMessage is supposed to block until a message is available.

In other words, I think if this find_message fails, we shouldn't return 1 below without filling in a message.

I think I will merge this and add a TODO about it.

Reply:
It should be blocked until there’s a matched message, but implementing this isn’t easy. Perhaps it can be done as a busy loop initially and then optimized later?